### ISSUE TITLE
Tools: How to handle examples for tools that need API keys?

### ISSUE THREAD

**Issue by AmeliaBR** [Maps4HTML/HTML-Map-Element-UseCases-Requirements#24]
I'd like to include pages of working demos to show how the various reference JS map widgets can (or can't) be used to address the different use cases, and to explore the details of their particular capabilities (e.g., the accessibility of their interfaces).  [The example page for “Display a specific location on a map”](https://maps4html.github.io/HTML-Map-Element-UseCases-Requirements/examples/map-viewers/single-location.html) shows the start of the idea.

Problem: Many of the mapping libraries require an API key linked to a credit card, even if they allow some usage for free.

Options:

- only include free products in the examples
- ask organizations involved in the community group to donate API keys

Note: I don't expect the actual page view counts on the examples to ever get very high. We could probably cap usage of most or all the APIs to their free tiers without worrying about hitting the limits. But, the links would still break eventually if no one renews the accounts they are associated with.

**Comment by prushforth** [comment]
I'm pretty happy with the open source option.  I especially appreciate Open Street Map content, although they make their centralized content available as services only by virtue of donations to their foundation; they prefer it, especially if your traffic gets too large, that you download their data and serve your own tiles.

There are other free tile providers that can be used, including the GoC's.

When it comes to services like MapBox and Google Maps etc., I feel that if they joined the community group it might be less risky, especially if they donated use of an API key to the group, to use their APIs as examples of what might be standardizable.

OTOH, there is a very well established pair of generic open source mapping APIs, in the form of Leaflet and OpenLayers, that only require access to data sources in order to function and which can certainly give us ideas for a standard API.  The issue with all these APIs is that we need to take the concept a step further and apply their functions to what a native DOM API could do vs what should reasonably be built as an abstraction on top of that DOM.  Currently as I've pointed out many times, possibly to the detriment of the message :-), *everything* in mapping is an abstraction built on current native DOM APIs.

In short my opinion is let's proceed in the open / free, and if orgs who have commercial mapping APIs want to join, and if they think it's important that features they have be considered, by all means let's collaborate.  Standards are defined by those who show up, after all.

**Comment by josejoserojas** [comment]
(Disclosure: I work for TomTom)
If you think your demos won't go over the 2500 transaction/per day limit, you can always use the [TomTom Maps APIs](https://developer.tomtom.com) 

You can get an API KEY without a credit card and you will always have support from us.

Take a look at our [Web SDK based tutorials](https://developer.tomtom.com/maps-sdk-web/functional-examples#map-vector-basic).
